src/app/core/config/validation-schemes/security-group-templates.scheme.json

```diff
@@ -0,0 +1,100 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "securityGroupTemplates",
+  "definitions": {
+    "TcpUdpPort": {
+      "type": "number",
+      "minimum": 0,
+      "maximum": 65535
+    },
+    "TcpUdpRule": {
+      "type": "object",
+      "required": ["ruleid", "protocol", "startport", "endport", "cidr"],
+      "additionalProperties": false,
+      "properties": {
+        "ruleid": {
+          "type": "string"
+        },
+        "protocol": {
+          "type": "string",
+          "enum": ["tcp", "udp"]
+        },
+        "startport": {
+          "$ref": "#/definitions/TcpUdpPort"
+        },
+        "endport": {
+          "$ref": "#/definitions/TcpUdpPort"
+        },
+        "cidr": {
+          "type": "string"
+        }
+      }
+    },
+    "IcmpRule": {
+      "type": "object",
+      "required": ["ruleid", "protocol", "icmpcode", "icmptype", "cidr"],
+      "additionalProperties": false,
+      "properties": {
+        "ruleid": {
+          "type": "string"
+        },
+        "protocol": {
+          "type": "string",
+          "const": "icmp"
+        },
+        "icmpcode": {
+          "type": "number"
+        },
+        "icmptype": {
+          "type": "number"
+        },
+        "cidr": {
+          "type": "string"
+        }
+      }
+    },
+    "RulesArray": {
+      "type": "array",
+      "uniqueItemProperties": [
+        "ruleid"
+      ],
+      "items": {
+        "anyOf": [
+          {
+            "$ref": "#/definitions/TcpUdpRule"
+          },
+          {
+            "$ref": "#/definitions/IcmpRule"
+          }
+        ]
+      }
+    }
+  },
+  "type": "array",
+  "uniqueItemProperties": ["id"],
```

Comment:
Add proper error message for this case.
Also, it will be more readable if you put definitions to the bottom.